### PR TITLE
ingest: compile mustache template only if field includes '{{''

### LINKED ISSUE #37120: ingest: only compile templates if the value is templated
Many of the ingest node processors support [template snippets](https://www.elastic.co/guide/en/elasticsearch/reference/current/accessing-data-in-pipelines.html#accessing-template-fields) (mustache).  Each field that supports [template snippets](https://www.elastic.co/guide/en/elasticsearch/reference/current/accessing-data-in-pipelines.html#accessing-template-fields) requires compilation the first time the pipeline is used. 

Elasticsearch has circuit breaker such that no more than 75 scripts per 5 minutes may be compiled. Every script (not only ingest node scripts) is subject to this ([configurable](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting-using.html#prefer-params)) limit. A moderate number of processors can trip this limit, even if no templated values are defined. 

For example, a single [set processor](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-processor.html) always counts as 2 compilations since the `field` and `value` both support [template snippets](https://www.elastic.co/guide/en/elasticsearch/reference/current/accessing-data-in-pipelines.html#accessing-template-fields).  The compilation happens even if the `field` and `value` do _not_ define a `{{ }}` template. 

This issue is to address the unnecessary compilation for fields that support [template snippets](https://www.elastic.co/guide/en/elasticsearch/reference/current/accessing-data-in-pipelines.html#accessing-template-fields). The goal to reduce the number of compilations to only the minimum number actually required. 

This issue is more likely on newer versions of Elasticsearch since in 6.4 additional fields were added to support template snippets and 6.5 the `if` condition was added, which also requires a script complication (if defined).

### PR DESCRIPTION
**Pull request by jakelandis**
Prior to this change, any field in an ingest node processor that supports
script templates would be compiled as mustache template regardless if they
contain a template or not. Compiling normal text as mustache templates is
harmless. However, each compilation counts against the script compilation
circuit breaker. A large number of processors without any templates or scripts
could un-intuitively trip the too many script compilations circuit breaker.

This change simple checks for '{{' in the text before it attempts to compile.

fixes #37120

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/ingest/ValueSourceTests.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 package org.elasticsearch.ingest;
 
+import org.elasticsearch.script.ScriptService;
+import org.elasticsearch.script.TemplateScript;
```
Comment (martijnvg) [inline server/src/test/java/org/elasticsearch/ingest/ValueSourceTests.java]: unused import

File: server/src/test/java/org/elasticsearch/ingest/ConfigurationUtilsTests.java [hunk 2]
```diff
@@ -36,7 +37,13 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
 import static org.hamcrest.Matchers.sameInstance;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
```
Comment (martijnvg) [inline server/src/test/java/org/elasticsearch/ingest/ConfigurationUtilsTests.java]: unused import

**Review by martijnvg — APPROVED** [review]
:+1: - there are two unused imports that make the pr build fail, but other than that LGTM.

**Comment by jakelandis** [comment]
thanks @martijnvg !  unused imports removed. will wait for green to merge.

**Comment by jakelandis** [comment]
@elasticmachine run gradle build tests 1

**Comment by jakelandis** [comment]
@elasticmachine run gradle build tests 1

**Comment by jakelandis** [comment]
@elasticmachine run gradle build tests 1
@elasticmachine run gradle build tests 2

**Pull request merged**